pkg/controller/ec2/instance/controller.go

```diff
@@ -221,6 +175,81 @@ func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.E
 	}, nil
 }
 
+func (e *external) describeInstance(ctx context.Context, instanceId string) (
+	*types.Instance,
+	awsec2.DescribeInstanceAttributeOutput,
+	error,
+) {
+	eg := errgroup.Group{}
+
+	var describeOutput *awsec2.DescribeInstancesOutput
+	var describeError error
+	eg.Go(func() error {
+		describeOutput, describeError = e.client.DescribeInstances(ctx, &awsec2.DescribeInstancesInput{
+			InstanceIds: []string{instanceId},
+		})
+		return nil
+	})
+
+	attrs := awsec2.DescribeInstanceAttributeOutput{}
+	descAttr := func(attr types.InstanceAttributeName) (*awsec2.DescribeInstanceAttributeOutput, error) {
+		return e.client.DescribeInstanceAttribute(ctx, &awsec2.DescribeInstanceAttributeInput{
+			InstanceId: &instanceId,
+			Attribute:  attr,
+		})
+	}
+
+	eg.Go(func() error {
+		if res, err := descAttr(types.InstanceAttributeNameDisableApiTermination); err != nil {
+			return errorutils.Wrap(err, "fetching DisableApiTermination")
+		} else {
+			attrs.DisableApiTermination = res.DisableApiTermination
+			return nil
+		}
+	})
```

Comment:
This is still quite a lot of duplicate code. Can we do it like in https://github.com/crossplane-contrib/provider-aws/pull/2047/files#r1577643597?

Also, there is no need for `else` if the previous `if` block had a `return`.

Reply:
> This is still quite a lot of duplicate code.

This is just the way error handling works in go, all these `if err != nil`. `descAttr` is straightforward and has no side-effects now. Having error handling inside `descAttr` adds a side-effect, and creation of a goroutine inside a small helper method. This makes things more complicated. And it would have same or more LOC.

> Also, there is no need for else if the previous if block had a return.

`res` is not visible outside of `if ... else`.